sound/soc/sof/intel/hda-dai.c

```diff
@@ -626,6 +626,7 @@ static int ipc4_be_dai_trigger(struct snd_pcm_substream *substream,
 		struct snd_sof_widget *pipe_widget = swidget->pipe_widget;
 		struct sof_ipc4_pipeline *pipeline = pipe_widget->private;
 
+		sdev = snd_soc_component_get_drvdata(swidget->scomp);
```

Comment:
You probably want a comment to explain the change - likely that the ALH/SoundWire dais are registered by different components for each link.

Reply:
> You probably want a comment to explain the change - likely that the ALH/SoundWire dais are registered by different components for each link.

Commit message updated.

Reply:
Hrm, but we can have only _one_ sdev, no?

Reply:
> Hrm, but we can have only _one_ sdev, no?

Yes, and we can get the only one sdev from swidget->scomp